src/actions/transactionSigner.js

```diff
@@ -18,5 +23,51 @@ export function setSecrets(secrets) {
   return {
     type: SET_SECRETS,
     secrets,
-  }
+  };
 }
+
+export const LEDGER_WALLET_SIGN_START = 'LEDGER_WALLET_SIGN_START';
+export const LEDGER_WALLET_SIGN_SUCCESS = 'LEDGER_WALLET_SIGN_SUCCESS';
+export const LEDGER_WALLET_SIGN_ERROR = 'LEDGER_WALLET_SIGN_ERROR';
+
+const DEFAULT_BIP = "44'/148'/0'";
+
+export function signWithLedger(txXDR) {
+  return dispatch => {
+    dispatch({ type: LEDGER_WALLET_SIGN_START });
+
+    let ledgerApi = new StellarLedger.Api(new StellarLedger.comm(120));
+    let transaction = new Transaction(txXDR);
+
+    let onError = err => {
+      dispatch({ 
+        type: LEDGER_WALLET_SIGN_ERROR,
+        error: err,
+       }); 
+    };
+
+    let onConnect = () => {
+      console.log(Promise)
+      BP.all([
+        ledgerApi.getPublicKey_async(DEFAULT_BIP),
+        ledgerApi.signTx_async(DEFAULT_BIP, transaction),
```

Comment:
This will sign only txs with one operation. We need to use `signTxHash_async` to sign any transaction. This should be probably handled in `stellar-ledger-api` (if one operation => `signTx_async` else `signTxHash_async`).

Also because of this we should display transaction hash in the transaction summary table.

Reply:
Created this issue: https://github.com/lenondupe/stellar-ledger-api/issues/3

Reply:
will fix later

Reply:
@lenondupe released a new version of `stellar-ledger-api`. It's now OK to use this function with every transaction.